javascript/javascript3/week3/TASK3-Shopping cart/index.js

```diff
@@ -1,130 +1,149 @@
-const totalPrice = document.getElementById("total-price");
-
-class Currency {
-  constructor(name, rate) {
-    this.name = name;
-    this.rate = rate;
-  }
-}
-
-const USD = new Currency("USD", 7.5);
-const EUR = new Currency("EUR", 8.2);
-const HUF = new Currency("HUF", 46.2);
+(function () {
+  const defaultCurrency = "DKK";
+  const totalPrice = document.getElementById("total-price");
+
+  class Product {
+    constructor(name, price) {
+      this.name = name;
+      this.price = price;
+    }
 
-class Product {
-  constructor(name, price) {
-    this.name = name;
-    this.price = price;
+    async convertToCurrency(currency) {
+      await getCurrencyRate(this.price, currency);
+    }
   }
 
-  convertToCurrency(currency) {
-    console.log(this.price * currency.rate);
-  }
-}
+  class ShoppingCart {
+    constructor() {
+      this.products = [];
+    }
 
-class ShoppingCart {
-  constructor() {
-    this.products = [];
-  }
+    addProduct(product) {
+      this.products.push(product);
+    }
 
-  addProduct(product) {
-    this.products.push(product);
-  }
+    removeProduct(product) {
+      const indexOfProduct = this.products.indexOf(product);
+      if (indexOfProduct > -1) {
+        // create a temporary new array of products
+        const tempProducts = this.products.map((product) => product);
+        // remove item from temporary array
+        tempProducts.splice(indexOfProduct, 1);
+        // replace products with the temporary array
+        this.products = tempProducts;
+      } else {
+        console.log("Can't find item.");
+      }
+    }
 
-  removeProduct(product) {
-    const indexOfProduct = this.products.indexOf(product);
-    if (indexOfProduct > -1) {
-      // create a temporary new array of products
-      const tempProducts = this.products.map((product) => product);
-      // remove item from temporary array
-      tempProducts.splice(indexOfProduct, 1);
-      // replace products with the temporary array
-      this.products = tempProducts;
-    } else {
-      console.log("Can't find item.");
+    searchProduct(productName) {
+      return this.products.filter((product) =>
+        product.name.toLowerCase().includes(productName.toLowerCase()),
+      );
     }
-  }
 
-  searchProduct(productName) {
-    return this.products.filter((item) => item.name === productName);
-  }
+    getTotal() {
+      return this.products.reduce((result, currentProduct) => result + currentProduct.price, 0);
+    }
 
-  getTotal() {
-    let total = 0;
-    this.products.forEach((item) => {
-      total = item.price + total;
-    });
-    return total;
-  }
+    renderProducts() {
+      const fragment = new DocumentFragment();
+      const ul = document.createElement("UL");
+      fragment.appendChild(ul);
+      this.products.forEach((product) => {
+        createListElement(product);
+      });
+
+      function createListElement(productToList) {
```

Comment:
not a big fan of functions inside of functions. I know you're doing it for scope, but again: Typescript and `private `would also do the trick.